java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -1099,147 +1113,172 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_round(
   CATCH_STD(env, 0);
 }
 
-JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_year(JNIEnv* env, jclass, jlong input_ptr)
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_extractDateTimeComponent(JNIEnv* env,
+                                                                                jclass,
+                                                                                jlong input_ptr,
+                                                                                jint component)
 {
   JNI_NULL_CHECK(env, input_ptr, "input is null", 0);
   try {
     cudf::jni::auto_set_device(env);
     cudf::column_view const* input = reinterpret_cast<cudf::column_view*>(input_ptr);
-    return release_as_jlong(cudf::datetime::extract_year(*input));
+    cudf::datetime::datetime_component comp;
+    switch (component) {
+      case 0: comp = cudf::datetime::datetime_component::YEAR; break;
+      case 1: comp = cudf::datetime::datetime_component::MONTH; break;
+      case 2: comp = cudf::datetime::datetime_component::DAY; break;
+      case 3: comp = cudf::datetime::datetime_component::WEEKDAY; break;
+      case 4: comp = cudf::datetime::datetime_component::HOUR; break;
+      case 5: comp = cudf::datetime::datetime_component::MINUTE; break;
+      case 6: comp = cudf::datetime::datetime_component::SECOND; break;
+      case 7: comp = cudf::datetime::datetime_component::MILLISECOND; break;
+      case 8: comp = cudf::datetime::datetime_component::MICROSECOND; break;
+      case 9: comp = cudf::datetime::datetime_component::NANOSECOND; break;
```

Comment:
This reads well.  I would have been tempted to use a calculation instead:
```c++
if ( component == std::clamp(component, YEAR, NANOSECOND) ) {
  comp = YEAR + component;
}
else { 
  JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "Invalid component", 0);
}
```
I think the explicit switch-case reads better.

Reply:
I get nervous that someone is going to change the order of things in the enum and break us, like add a new one. In other places we just cast the int/byte/whatever to the enum and it generally works, but it is too brittle. Like what happens if they want to add quarter to this interface? would they add it to the end? Or in between year and month?

Reply:
Yep, agreed.  I think I should lean towards your approach, should I have to make a similar choice henceforth.